src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests_TestMarkup.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Classification;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Remote.Testing;
+using Microsoft.CodeAnalysis.Shared;
+using Microsoft.CodeAnalysis.Test.Utilities.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+using static Microsoft.CodeAnalysis.Editor.UnitTests.Classification.FormattedClassifications;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Classification;
+
+public partial class SemanticClassifierTests : AbstractCSharpClassifierTests
+{
+    private const string s_testMarkup = """
+
+        static class Test
+        {
+            public static void M([System.Diagnostics.CodeAnalysis.StringSyntax("C#-test")] string code) { }
+        }
+        """ + EmbeddedLanguagesTestConstants.StringSyntaxAttributeCodeCSharp;
+
+    protected async Task TestEmbeddedCSharpAsync(
+       string code,
+       TestHost testHost,
+       params FormattedClassification[] expected)
+    {
+        var allCode = $$"""""
+            class C
+            {
+                void M()
+                {
+                    Test.M(""""
+            {{code}}
+            """");
+                }
+            }
+            """"" + s_testMarkup;
```

Comment:
note the structure here.  The test cases below have teh code that will be in the string literal here that is passed to `Test.M` (which is marked with the `C#-test` lang attribute.  So the tests below are testing how *embedded* c# is classified, not normal c#.